src/lib/skilling/skills/hunter/rumours/rumours.ts

```diff
@@ -0,0 +1,176 @@
+import { Time, roll } from 'e';
+import { QuestID } from '../../../../minions/data/quests';
+import type { RumourActivityTaskOptions } from '../../../../types/minions';
+import addSubTaskToActivityTask from '../../../../util/addSubTaskToActivityTask';
+import { calcMaxTripLength } from '../../../../util/calcMaxTripLength';
+import { type Creature, HunterTechniqueEnum, SkillsEnum } from '../../../types';
+import creatures from '../creatures';
+import { getRumourBlockList } from './rumourBlocks';
+import { type Rumour, type RumourOption, tierToHunterLevel } from './util';
+
+const hunterTechniqueToRate: { [key in HunterTechniqueEnum]: number } = {
+	[HunterTechniqueEnum.AerialFishing]: -1,
+	[HunterTechniqueEnum.DriftNet]: -1,
+	[HunterTechniqueEnum.BirdSnaring]: 20,
+	[HunterTechniqueEnum.BoxTrapping]: 50,
+	[HunterTechniqueEnum.ButterflyNetting]: 75,
+	[HunterTechniqueEnum.DeadfallTrapping]: 15,
+	[HunterTechniqueEnum.Falconry]: 10,
+	[HunterTechniqueEnum.MagicBoxTrapping]: -1,
+	[HunterTechniqueEnum.NetTrapping]: 25,
+	[HunterTechniqueEnum.PitfallTrapping]: 15,
+	[HunterTechniqueEnum.RabbitSnaring]: -1,
+	[HunterTechniqueEnum.Tracking]: 15
+};
+
+export async function rumoursCommand(userID: string, channelID: string, input?: RumourOption): CommandResponse {
+	const user = await mUserFetch(userID);
+	const hunterLevel = user.skillsAsLevels.hunter;
+
+	if (!user.user.finished_quest_ids.includes(QuestID.ChildrenOfTheSun)) {
+		return `You need to complete the "Children of the Sun" quest before you can be assigned a hunter rumour. Send your minion to do the quest using: ${mentionCommand(
+			globalClient,
+			'activities',
+			'quest'
+		)}.`;
+	}
+
+	if (hunterLevel < 46) {
+		return 'You need at least level 46 in Hunter to begin doing Hunter Rumours.';
+	}
+
+	if (!input) {
+		const bestTier = highestLevelRumour(user);
+		return rumoursCommand(user.id, channelID, bestTier);
+	}
+
+	if (hunterLevel < tierToHunterLevel[input]) {
+		return `You need level ${tierToHunterLevel[input]} hunter to do rumours of ${input} difficulty!`;
+	}
+
+	const maxTripLength = calcMaxTripLength(user);
+	const Rumours: Rumour[] = await generateRumourTasks(user, input, maxTripLength);
+
+	const totalDuration = Rumours.reduce((acc, rumour) => acc + rumour.duration, 0);
+
+	await addSubTaskToActivityTask<RumourActivityTaskOptions>({
+		userID: user.id,
+		channelID: channelID,
+		quantity: Rumours.length,
+		tier: input,
+		rumours: Rumours,
+		duration: totalDuration,
+		type: 'Rumour'
+	});
+
+	return `${user.minionName} is now completing ${input} tier rumours. It'll return in ${formatDuration(totalDuration)}.`;
+}
+
+async function generateRumourTasks(user: MUser, tier: RumourOption, maxLength: Time.Minute) {
+	const Rumours: Rumour[] = [];
+	let totalDuration = 0;
+	const maxDurationInSeconds = maxLength / 1000;
+	const blockList = (await getRumourBlockList(user.id)).rumour_blocked_ids;
```

Comment:
Cant you just do `user.user.rumour_blocked_ids` ? the `getRumourBlockList` function seems pointless